cirq/ops/common_channels.py

```diff
@@ -187,3 +189,603 @@ def depolarize(p: float) -> DepolarizingChannel:
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PD({!r})'.format(self._gamma)
+
+
+def phase_damp(gamma: float) -> PhaseDampingChannel:
+    """
+    Creates a PhaseDampingChannel with damping rate gamma
+
+    This channel evolves a density matrix via
+           \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \begin{bmatrix}
+                1 & 0  \\
+                0 & \sqrt{1 - \gamma}
+              \end{bmatrix}
+        M_1 = \begin{bmatrix}
+                0 & 0 \\
+                0 & \sqrt{\gamma}
+              \end{bmatrix}
+
+    Args:
+        gamma: The damping rate
+
+    Raises:
+        ValueError is gamma is not a valid probability.
+    """
+    return PhaseDampingChannel(gamma)
+
+
+class PhaseFlipChannel(raw_types.Gate):
+    """
+    Flips the sign of the phase of a qubit with a given probability.
+    """
+
+    def __init__(self, p) -> None:
+
+        """
+        Construct a channel to flip the phase with probability p
+
+        This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+        With
+            M_0 = \sqrt{p} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & 1
+                            \end{bmatrix}
+            M_1 = \sqrt{1-p} \begin{bmatrix}
+                                1 & 0 \\
+                                0 & -1
+                            \end{bmatrix}
+
+        Args:
+            p: the probability of a phase flip
+
+        Raises:
+            ValueError if p is not a valid probability.
+        """
+
+        def validate_probability(p, p_str):
+            if p < 0:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._p = validate_probability(p, 'p')
+        self._delegate = AsymmetricDepolarizingChannel(0., 0., 1. - p)
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        kraus_ops = list(self._delegate._channel_())
+        # just return identity and z term
+        return (kraus_ops[0], kraus_ops[3])
+
+    def _eq_tuple(self):
+        return (PhaseFlipChannel, self._p)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.phase_flip(p={!r})'.format(self._p)
+
+    def __str__(self) -> str:
+        return 'phase_flip(p={!r})'.format(self._p)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PF({!r})'.format(self._p)
+
+
+def phase_flip(p: float) -> PhaseFlipChannel:
```

Comment:
Have `p` default to `None`, change return type to `Union[ops.ZPowGate, PhaseFlipChannel]`, declare an overload with no argument returning `ops.ZPowGate` and an overload with the non-optional argument returning `PhaseFlipChannel`. If `p is None` then return `cirq.Z`.